### PR TITLE
pack: escape characters before using them in glob patterns

### PR DESCRIPTION
**Pull request by bk2204**
When we open a pack file, we use a glob pattern to find the pack files that are in our pack directory and then attempt to open them.  This works fine if our directory contains no characters that are special to the shell, but works poorly when we handle characters that are special in glob patterns, most notably the left square bracket.  In such a case, our pattern doesn't actually match anything, so we don't open any pack files.  Git LFS then complains about a missing object.

Escape all the characters which are special to the shell so that we can handle these paths properly.  Use character classes instead of backslashes because backslash escaping is not available on Windows.  For the same reason, punt on handling backslashes on Unix systems because there isn't a great way to handle them portably.

### CODE REVIEW

File: pack/set.go [hunk 1]
```diff
@@ -91,6 +92,21 @@ func NewSet(db string) (*Set, error) {
...
+// backslash escapes on Windows.
+var globEscapes = map[string]string{
+	"*": "[*]",
+	"?": "[?]",
+	"[": "[[]",
+}
+
+func escapeGlobPattern(s string) string {
+	for char, escape := range globEscapes {
+		s = strings.Replace(s, char, escape, -1)
```
Comment (ttaylorr) [inline pack/set.go]: Nit: in newer versions of Go, I believe that `strings.ReplaceAll` is preferred.

File: pack/set.go [hunk 2]
```diff
@@ -40,7 +41,7 @@ var (
 func NewSet(db string) (*Set, error) {
 	pd := filepath.Join(db, "pack")
 
-	paths, err := filepath.Glob(filepath.Join(pd, "*.pack"))
+	paths, err := filepath.Glob(filepath.Join(escapeGlobPattern(pd), "*.pack"))
```
Comment (ttaylorr) [inline pack/set.go]: Do you think that this change is test-able in gitobj? If not, I think either the integration test(s) in LFS, or a "unit" test that creates a repository might be appropriate.

File: pack/set.go (hunk 1 above)
Comment (bk2204) [reply pack/set.go]: Yes, but I think this repository still uses Travis, which uses Go 1.11.  `strings.ReplaceAll` is new in 1.12.  I've been bitten by this in the past (multiple times, even).

File: pack/set.go (hunk 2 above)
Comment (bk2204) [reply pack/set.go]: I don't think this is testable in `gitobj`.  We don't have any test code that calls `NewSet`, only some of the more internal functions.

As for an integration test in Git LFS, I have already written such a test, have already verified it works, and will include it when I bump the version of `gitobj` in the main repository.

**Pull request merged**